packages/babel-traverse/src/path/introspection.ts

```diff
@@ -181,7 +181,9 @@ export function referencesImport(
 ): boolean {
   if (!this.isReferencedIdentifier()) {
     if (
-      (this.isMemberExpression() || this.isOptionalMemberExpression()) &&
+      (this.isMemberExpression() ||
+        this.isJSXMemberExpression() ||
+        this.isOptionalMemberExpression()) &&
       (this.node.computed
         ? isStringLiteral(this.node.property, { value: importName })
         : (this.node.property as t.Identifier).name === importName)
```

Comment:
I tried moving `this.isJSXMemberExpression()` outside of the existing `&&` expression, but `this.node` type resolves to `never` for some reason

```ts
(this.isJSXMemberExpression() &&
  (this.node.property as t.JSXIdentifier).name === importName))
        ^
        never
```

Reply:
Doing it like this isn't good, thought. `this.isJSXMemberExpression()` basically disables type checking for the rest of the expression.

Reply:
Ok, so `JSXMemberExpression` is not a part of `Expression` union type in `babel-types/src/ast-types/generated`. Searching for the place to add it.